pkgs/applications/science/math/mathematica/default.nix

```diff
@@ -93,6 +94,12 @@ stdenv.mkDerivation rec {
 
     echo "=== Running MathInstaller ==="
     ./MathInstaller -auto -createdir=y -execdir=$out/bin -targetdir=$out/libexec/Mathematica -silent
+
+    # Fix library paths
```

Comment:
In the preFixup the rpath is already being fixed. Adding `zlib` to `buildInputs` should be sufficient. Please test.